src/components/structures/RoomView.js

```diff
@@ -95,15 +103,14 @@ module.exports = React.createClass({
 
     getInitialState: function() {
         var room;
+        var room_id;
         if (this.props.roomAddress[0] == '!') {
-            room = MatrixClientPeg.get().getRoom(this.props.roomAddress);
-        } else {
-            room = MatrixTools.getRoomForAlias(
-                MatrixClientPeg.get().getRooms(), this.props.roomAddress
-            );
+            room_id = this.props.roomAddress;
```

Comment:
would it make more sense to initialise `room_id` and `room` to null, and put this logic in `componentWillMount`, to keep it together with the alias case?